generateDeps.py

```diff
@@ -0,0 +1,49 @@
+import os
+
+class GenerateDeps():
+
+    template = """#Dependencies required by NovelRT on Windows/macOS
+set(DEPENDENCY_FILES
+#DEP#
+)
+
+foreach(dependency ${DEPENDENCY_FILES})
+  configure_file(${dependency} ${dependency} COPYONLY)
+endforeach()
+
+add_library(ThirdParty INTERFACE IMPORTED GLOBAL)
+set_property(TARGET ThirdParty PROPERTY RESOURCE ${DEPENDENCY_FILES})
+if(WIN32 OR APPLE)
+  set(ext "")
+  if(WIN32)
+    set(ext "*.dll")
+  else()
+    set(ext "*.dylib")
+  endif()
+  install(FILES ${DEPENDENCY_FILES}
+    DESTINATION bin
+  )
+endif()"""
+
+    def CreateCMakeLists(self, depPath, cmakePath):
+        files = os.scandir(depPath)
+        safeExts = [".dll",".so",".a",".dylib"]
```

Comment:
.so/.a files likely don't need to be copied since they're handled by RUNPATH/RPATH (and .a files are basically just tarballs of .o files, which are linked directly into the target library/application anyway)